client/actions/events/api.js

```diff
@@ -1101,46 +1101,59 @@ const _saveLocation = (event) => (
+            Promise.resolve({}) :
+            dispatch(self.fetchById(eventUpdates._id, {saveToStore: false, loadPlanning: false}))
+    )
+        .then((originalEvent) => {
+            // clone the original because `save` will modify it
+            const original = cloneDeep(originalEvent);
+
+            // clone the updates as we're going to modify it
+            let updates = cloneDeep(eventUpdates);
+
+            // remove links if it contains only null values
+            if (updates.links && updates.links.length > 0) {
+                updates.links = updates.links.filter((l) => (l));
+                if (!updates.links.length) {
+                    delete updates.links;
+                }
             }
-        }
-        // retrieve original
-        let original = selectors.getEvents(getState())[newEvent._id];
-        // clone the original because `save` will modify it
 
-        original = cloneDeep(original) || {};
-        let clonedEvent = cloneDeep(newEvent) || {};
+            // save the timezone. This is useful for recurring events
+            if (updates.dates) {
+                updates.dates.tz = moment.tz.guess();
+            }
 
-        // save the timezone. This is useful for recurring events
-        if (clonedEvent.dates) {
-            clonedEvent.dates.tz = moment.tz.guess();
-        }
+            original.location = original.location ? [original.location] : null;
+            updates.location = updates.location ? [updates.location] : null;
 
-        original.location = original.location ? [original.location] : null;
-        clonedEvent.location = clonedEvent.location ? [clonedEvent.location] : null;
+            // remove all properties starting with _
+            // and updates that are the same as original
+            updates = pickBy(updates, (v, k) => (
+                !k.startsWith('_') &&
+                !isEqual(updates[k], original[k])
+            ));
 
-        // remove all properties starting with _,
-        // otherwise it will fail for "unknown field" with `_type`
-        clonedEvent = pickBy(clonedEvent, (v, k) => (
-            !k.startsWith('_') &&
-            !isEqual(clonedEvent[k], original[k])
-        ));
+            updates.update_method = get(updates, 'update_method.value', EventUpdateMethods[0].value);
 
-        clonedEvent.update_method = get(clonedEvent, 'update_method.value', EventUpdateMethods[0].value);
+            return api('events').save(original, updates);
+        })
+        .then((data) => {
+            // return a list of events (can have several because of reccurence)
+            const events = get(data, '_items') || [data];
+
+            // If this Event has been created from a Planning item
+            // then link the new Event to it
+            if (!get(eventUpdates, '_id') && get(eventUpdates, '_planning_item') && get(events, '[0]._id')) {
+                return dispatch(planningApi.linkToEvent(eventUpdates._planning_item, events[0]._id))
```

Comment:
Would it be better to achieve this in the backend when we create the event instead of another call from planning api?

Reply:
Good call, I'll make the changes